alf/utils/video_recorder.py

```diff
@@ -140,16 +152,46 @@ def __init__(self,
         if frames_per_sec is not None:
             self.frames_per_sec = frames_per_sec  # overwrite the base class
 
-    def capture_frame(self, pred_info=None):
+        self._future_steps = future_steps
+        self._frame_buffer = []
+        self._observation_buffer = []
+        self._reward_buffer = []
+        self._action_buffer = []
```

Comment:
Organizing these variables into a namedtuple would be better:
```
RecorderBuffers = namedtuple("RecorderBuffers", ["frame", "observation", "reward", "action"], default_value=[])

Reply:
Improved